internal/controller/scheduler/scheduler.go

```diff
@@ -75,6 +76,7 @@ func New(logger *zap.Logger, client kubernetes.Interface, cfg Config) *worker {
 }
 
 type KubernetesPlugin struct {
+	Job               *batchv1.Job           `json:"job,omitempty"`
```

Comment:
Using `*batchv1.Job` here could get confusing - someone might try to write:

```yaml
plugins:
- kubernetes:
    job:
      template:
        spec:
          containers:
            ...
```

which wouldn't be rejected for any failure to parse, and the user might expect that defining containers that way would work but instead it would be silently ignored. 

(Maybe one day the whole `KubernetesPlugin` should be changed to revolve around a `*batchv1.Job` instead of defining pieces of it. Needs thought.)

In the meantime I would suggest either just adding `JobActiveDeadlineSeconds` as a `KubernetesPlugin` field directly, or maybe wrapping it in a new `Job` struct that can be grown over time to include other job parameters.

Reply:
This is step 1 of a long journey to fully transition to `*batchv1.Job`. I went back and forth on whether to implement it this way and possibly cause confusion for users that try to leverage the entire Job spec. There are additional tickets we have that will continue to build off of the Job spec (`backoffLimit`, `podFailurePolicy`, etc.) bit by bit. We could implement this (and the future additional fields) as fields in the plugin and eventually bring them back into the full Job spec when the time comes to more precisely control the feature set.